spec/10-expressions.md

```diff
@@ -810,8 +819,9 @@ passing it the optional *argument-expression-list*. If the class has no
 constructor, the constructor that class inherits (if any) is used. The class
 can also specify no constructor definition, in this case the constructor call is omitted.
 
-The result of an *object-creation-expression* is an object
-of the type specified by *class-type-designator*.
+The result of a named-type *object-creation-expression* is an object of the type specified by *class-type-designator*. The result of an anonymous class object-creation-expression is an object of unspecified type. 
```

Comment:
That's not entirely correct - while the type is not fully specified, it is known that it satisfies class-base-clause and class-interface-clause.